app/src/main.rs

```diff
@@ -66,12 +78,17 @@ fn main() {
 
     let mut retval = sgx_status_t::SGX_SUCCESS;
 
+    let binary = fs::read(WASM_FILE).unwrap();
+
+    let mut result_out = 0i32;
+
     let result = unsafe {
-        ecall_test(
+        exec_wasm_test(
             enclave.geteid(),
             &mut retval,
-            input_string.as_ptr() as *const u8,
-            input_string.len(),
+            binary.as_ptr() as _,
```

Comment:
I were not aware that the general recommendation were to use explicit types in this case. 

The reason for using `_` is that the type can be very easily inferred by looking at the function signature of the function being called, but I will update it to be explicit to follow the recommendation from the docs.

Reply:
I can remove the `as` casting here since `&T` and `&mut T` will be [coerced](https://doc.rust-lang.org/std/primitive.pointer.html#1-coerce-a-reference-t-or-mutable-reference-mut-t) into `*const T` and `*mut T`.